### PR TITLE
fix(pages): Prevent public "Missing access level" in write access field

### LINKED ISSUE #8905: Pages Write Access: Missing Access Level Name
Newbie here, and just started working on Elgg and Git a few days ago. Although this looks like a simple fix, I don't want to change anything because I am still learning how this all works and don't want to break anything. 

When creating a page, the write access has four levels; private, friends, logged in users and "missing access level name". I think this is supposed to be public. 

screen shot 2015-09-03 at 9 00 27 am

### PR DESCRIPTION
**Pull request by beck24**
This was previously being populated by default access on a non-edit form

Fixes #8905

### CODE REVIEW

**Comment by mrclay** [comment]
LGTM

File: mod/pages/start.php [hunk 1]
```diff
@@ -427,3 +430,29 @@ function pages_write_access_options_hook($hook, $type, $return_value, $params) {
...
+
+
+/**
+ * Called on view_vars, input/access hook
+ * Prevent ACCESS_PUBLIC from ending up as a write access option
+ * 
+ * @param type $hook
+ * @param type $type
+ * @param type $return
+ * @param type $params
```
Comment (juho-jaakkola) [inline mod/pages/start.php]: Let's add the correct types.

File: mod/pages/start.php [hunk 2]
```diff
@@ -427,3 +430,29 @@ function pages_write_access_options_hook($hook, $type, $return_value, $params) {
...
+ * @return string
+ */
+function pages_write_access_vars($hook, $type, $return, $params) {
+	if ($return['name'] == 'write_access_id' && $return['purpose'] = 'write') {
```
Comment (juho-jaakkola) [inline mod/pages/start.php]: Would be easier to read if the logic was less nested.

```php
if ($return['name'] !== 'write_access_id') {
    return $return;
}
```
etc.

**Comment by beck24** [comment]
refactored for readability

**Comment by hypeJunction** [comment]
LGTM

**Pull request merged**